contracts/stork_pyth_adapter/contracts/StorkPythAdapter.sol

```diff
@@ -0,0 +1,167 @@
+
+pragma solidity >=0.8.24 <0.9.0;
+
+import "@pythnetwork/pyth-sdk-solidity/IPyth.sol";
+import "@pythnetwork/pyth-sdk-solidity/PythStructs.sol";
+
+/**
+ * @title A port of the IPyth interface that supports Stork price feeds
+ */
+contract StorkPythAdapter is IPyth {
+    IStork public stork;
+    int32 private exponent = -18;
+    uint64 private confidenceInterval = 0;
+
+    constructor(address _stork) {
+        stork = IStork(_stork);
+    }
+
+    function convertInt192ToInt64(int192 value) public pure returns (int64) {
+        require(value >= type(int64).min && value <= type(int64).max, "Overflow");
+        return int64(value);
+    }
+
+    /// @notice Returns the period (in seconds) that a price feed is considered valid since its publish time
+    function getValidTimePeriod() external view returns (uint validTimePeriod) {
+        return stork.validTimePeriodSeconds();
+    }
+
+    /// @notice Returns the price.
+    /// @dev Note that confidence intervals are not supported
+    /// @return price - please read the documentation of PythStructs.Price to understand how to use this safely.
+    function getPrice(
+        bytes32 id
+    ) external view returns (PythStructs.Price memory price) {
+        StorkStructs.TemporalNumericValue memory temporalNumericValue = stork.getTemporalNumericValueV1(id);
+        return PythStructs.Price(
+            convertInt192ToInt64(temporalNumericValue.quantizedValue),
+            confidenceInterval,
+            exponent,
+            temporalNumericValue.timestampNs / 1000000000
+        );
+    }
+
+    /// @notice Returns the price of a price feed without any sanity checks.
+    /// @dev Note that confidence intervals are not supported
+    /// @return price - please read the documentation of PythStructs.Price to understand how to use this safely.
+    function getPriceUnsafe(
+        bytes32 id
+    ) external view returns (PythStructs.Price memory price) {
+        StorkStructs.TemporalNumericValue memory temporalNumericValue = stork.getTemporalNumericValueUnsafeV1(id);
+        return PythStructs.Price(
+            convertInt192ToInt64(temporalNumericValue.quantizedValue),
+            confidenceInterval,
+            exponent,
+            temporalNumericValue.timestampNs / 1000000000
+        );
+    }
+
+    /// @notice Returns the price that is no older than `age` seconds of the current time.
+    /// @dev Note that confidence intervals are not supported
+    /// @return price - please read the documentation of PythStructs.Price to understand how to use this safely.
+    function getPriceNoOlderThan(
+        bytes32 id,
+        uint age
+    ) external view returns (PythStructs.Price memory price) {
+        StorkStructs.TemporalNumericValue memory temporalNumericValue = stork.getTemporalNumericValueUnsafeV1(id);
+
+        if (block.timestamp - (temporalNumericValue.timestampNs / 1000000000) > age) {
+            revert("Value is stale");
+        }
```

Comment:
copied this logic from `getTemporalNumericValueV1` since it's basically the same thing but with a variable `age`